src/wp-includes/media.php

```diff
@@ -5783,6 +5838,20 @@ function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
 		if ( wp_lazy_loading_enabled( $tag_name, $context ) ) {
 			$loading_attrs['loading'] = 'lazy';
 		}
+
+		// Do not override `decoding` if it is already set.
+		if ( 'img' === $tag_name ) {
+			$loading_attrs['decoding'] = 'async';
+			if ( array_key_exists( 'decoding', $attr ) ) {
+				$loading_attrs['decoding'] = $attr['decoding'];
+			}
+		}
+	}
+
+	// If the element is not in the viewport, and has not sizes, it should only have the decoding attribute.
```

Comment:
@felixarntz

this aims to have the same effect as the early bail of the previous comment but will include the `decoding` attribute.